opm/grid/CpGrid.hpp

```diff
@@ -555,58 +556,590 @@ namespace Dune
+                                           Dune::cpgrid::EntityVariableBase<cpgrid::Geometry<2,3>>& adapted_faces,
+                                           Dune::cpgrid::EntityVariableBase<enum face_tag>& mutable_face_tags,
+                                           Dune::cpgrid::EntityVariableBase<Dune::FieldVector<double,3>>& mutable_face_normals,
+                                           Opm::SparseTable<int>& adapted_face_to_point,
+                                           const int& face_count,
+                                           /* Leaf grid View Cells argumemts  */
+                                           Dune::cpgrid::EntityVariableBase<cpgrid::Geometry<3,3>>& adapted_cells,
+                                           std::vector<std::array<int,8>>& adapted_cell_to_point,
+                                           std::vector<int>& adapted_global_cell,
+                                           const int& cell_count,
+                                           cpgrid::OrientedEntityTable<0,1>& adapted_cell_to_face,
+                                           cpgrid::OrientedEntityTable<1,0>& adapted_face_to_cell,
+                                           /* Auxiliary arguments */
+                                           std::unordered_map<int,std::array<int,2>> adaptedCorner_to_elemLgrAndElemLgrCorner,
+                                           std::unordered_map<int,std::array<int,2>> adaptedFace_to_elemLgrAndElemLgrFace,
+                                           std::unordered_map<int,std::array<int,2>> adaptedCell_to_elemLgrAndElemLgrCell,
+                                           std::map<std::array<int,2>,int> elemLgrAndElemLgrFace_to_adaptedFace,
+                                           const std::vector<std::vector<std::pair<int, std::vector<int>>>>& faceInMarkedElemAndRefinedFaces,
+                                           Dune::cpgrid::DefaultGeometryPolicy adapted_geometries,
+                                           std::map<std::array<int,2>,int> elemLgrAndElemLgrCorner_to_adaptedCorner,
+                                           std::map<std::array<int,2>, std::array<int,2>> vanishedRefinedCorner_to_itsLastAppearance,
+                                           const std::vector<std::shared_ptr<Dune::cpgrid::CpGridData>>& markedElem_to_itsLgr,
+                                           const std::vector<int>& assignRefinedLevel,
+                                           std::map<std::array<int,2>,int> markedElemAndEquivRefinedCorn_to_corner,
+                                           const std::vector<std::vector<std::array<int,2>>>& cornerInMarkedElemWithEquivRefinedCorner,
+                                           const std::vector<std::array<int,3>>& cells_per_dim_vec,
+                                           int preAdaptMaxLevel);
+
+        /// @brief Get the ijk index of a refined corner, given its corner index of a single-cell-refinement.
```

Comment:
What ijk index is this? Is that the index of the refined cell with respect to the parent cell

Reply:
If we refine a single cell with index "cell_idx (or "elemLgr")" on the "starting grid" (preAdapt grid, grid where we select and mark elements fore refinement), with cells_per_dim = { nx, ny, nz}, then this single-cell-auxiliary-refinement  has(with auxiliary I mean that in the end, the grid does not store this info, it's only an auxiliary computation):
- (nx +1)*(ny +1)*(nz +1) (auxiliary) refined corners,
- ((nx+1)*ny*nz) + (nx*(ny+1)*nz) + (nx*ny*(nz+1)) (auxiliary) refined faces, and
- nx*ny*nz (auxiliary) refined cells. 

The function argument cornerIdxInLgr is then an integer between 0 and  (nx +1)*(ny +1)*(nz +1) .
The return ijk value is computed taking into account how refined corners are stored in Geometry::refineCellifiedPatch [the name of this method does not fit so well now that we are actually refining single cells]. The order of the created corners in the refine method in Geometry.hpp is the followed one:
(j*(nx+1)*(nz+1)) + (i*(nz+1)) + k.

I hope this helps! I try to improve the description in the documentation.

Reply:
Understood